k-distribution/src/main/scripts/lib/kit/__main__.py

```diff
@@ -0,0 +1,115 @@
+
+from .constrained_term import ConstrainedTerm
+from .prover import CfgProver
+from .utils import eliminate_generated_top
+
+
+def main():
+    argument_parser = create_argument_parser()
+    args = argument_parser.parse_args()
+
+    command = args.command
+    if command == 'init':
+        init(args)
+    elif command == 'prove':
+        prove(args)
+    else:
+        assert False
+
+
+def create_argument_parser() -> ArgumentParser:
+    argument_parser = ArgumentParser(prog='kit')
+    argument_parser.add_argument('-w', '--workdir', metavar='WORKDIR', dest='work_dir', type=Path, default=Path('.'), help='Working directory.')
+    command_subparsers = argument_parser.add_subparsers(dest='command', required=True)
+
+    init_subparser = command_subparsers.add_parser('init', help='Initialize prover.')
+    init_subparser.add_argument('kompiled_dir', type=dir_path, help='Path to *-kompiled directory.')
+    init_subparser.add_argument('main_file', type=file_path, help='Path to main definition file.')
+    init_subparser.add_argument('spec_file', type=file_path, help='Path to specification JSON file.')
+    init_subparser.add_argument('-I', '--include', metavar='INCLUDE', dest='include_dirs', type=dir_path, default=[], action='append', help='Path to directory to include.')
+    init_subparser.add_argument('-f', '--force', dest='force', default=False, action='store_true', help='Overwrite existing configuration.')
+
+    expand_subparser = command_subparsers.add_parser('expand', help='Expand node.')
+    expand_subparser.add_argument('node_id', type=str, help='ID of node to expland.')
+
+    # Dummy command for testing
+    command_subparsers.add_parser('prove', help='Load prover from config file.')
+
+    return argument_parser
+
+
+def init(args):
+    if not args.work_dir.exists():
+        raise ValueError(f'Working directory doe not exist: {args.work_dir}')
+
+    work_dir = args.work_dir / '.kit'
+    if not args.force and work_dir.exists():
+        raise ValueError(f'Summary already exists at: {work_dir}')
+
+    # Init CFG
+    cfg_file = work_dir / 'cfg.json'
+
+    claim = _extract_claim_from_spec_file(args.spec_file)
+    cfg = _create_cfg_from_claim(claim)
+
+    # Create config
+    config_file = work_dir / 'config.json'
+
+    config = {
+        'kompiledDir': relpath(args.kompiled_dir, work_dir),
+        'mainFile': relpath(args.main_file, work_dir),
+        'cfgFile': relpath(cfg_file, work_dir),
+        'includeDirs': [relpath(include_dir, work_dir) for include_dir in args.include_dirs],
+    }
+
+    work_dir.mkdir(exist_ok=True)
+
+    with open(config_file, 'w') as f:
+        f.write(json.dumps(config, indent=4))
+        f.write('\n')
+
+    with open(cfg_file, 'w') as f:
+        f.write(cfg.to_json())
+        f.write('\n')
+
+
+def _extract_claim_from_spec_file(spec_file: Path) -> KClaim:
+    with open(spec_file, 'r') as f:
+        spec = json.load(f)
+
+    module = KFlatModule.from_dict(spec['term'])
```

Comment:
Would it be possible to make it so we can say:

```suggestion
    module = KFlatModule(spec['term'])
```

instead?

Reply:
Or maybe this isn't a good idea, not sure.

Reply:
It is doable, though I think it is best to have a single, straightforward `__init__` function (that basically just provides values for each attribute) and static factory methods for all the rest.

Reply:
Only if you think ti makes sense and is "pythonic"